### ISSUE TITLE
Can't initialize integration after 2024.4

### ISSUE THREAD

**Issue by veista** [veista/nilan#123]
Integration will not load.

**Comment by jamespreedy** [comment]
Same, please let me know if I can provide any logs etc that could help. Thanks!

**Comment by veista** [comment]
Sorry, I don't have time to look into this right now. Could not fix it easily, there is some issue introduced to the modbus wrapper on HA after making it async.

I would hold on updating/roll back for now.

**Comment by AdamGrumlik** [comment]
2024.4.0 Integration not working:(

**Comment by jamespreedy** [comment]
@AdamGrumlik - see viesta's comment above.
He's the author of the integration, is aware of the issue with 2024.4 but doesn't have time to look at this at the moment.

Best advise at this point is to roll back for now if you've already updated and the Nilan integration is required. 👍🏻

**Comment by kfroeschl** [comment]
I think I've found the issue,

HA ModbusHub is now fully async:
https://github.com/home-assistant/core/commit/7cba34b2e6b53621e6188ff2b473a3553e72bdcd

and thus the `success = await self._modbus.async_setup()`([device.py](https://github.com/veista/nilan/blob/master/custom_components/nilan/device.py#L59)) call does no longer wait for the connection to be established.

in my local workaround i've added a `await asyncio.sleep(10)` right after to get it running again for now.
Though this is far from a beautiful solution it works for now.

(+ I've added the await before `await self._modbus.async_close()` to avoid another error)
